src/plasma/malloc.cc

```diff
@@ -33,15 +35,10 @@ int fake_munmap(void *, size_t);
 
 struct mmap_record {
   int fd;
-  void *pointer;
   int64_t size;
-  UT_hash_handle hh_fd;
-  UT_hash_handle hh_pointer;
 };
 
-/* TODO(rshin): Don't have two hash tables. */
-struct mmap_record *records_by_fd = NULL;
-struct mmap_record *records_by_pointer = NULL;
+std::unordered_map<void *, mmap_record> mmap_records;
 
 const int GRANULARITY_MULTIPLIER = 2;
```

Comment:
constexpr is probably better, and you can also put this in the unnamed namespace.